lib/internal/streams/transform.js

```diff
@@ -82,6 +83,26 @@ function Transform(options) {
   if (!(this instanceof Transform))
     return new Transform(options);
 
+  // TODO (ronag): This should preferably always be
+  // applied but would be semver-major. Or even better;
+  // make Transform a Readable with the Writable interface.
+  const readableHighWaterMark = options ? getHighWaterMark(this, options, 'readableHighWaterMark', true) : null;
+  if (readableHighWaterMark === 0) {
+    // A Duplex will buffer both on the writable and readable side while
+    // a Transform just wants to buffer hwm number of elements. To avoid
+    // buffering twice we disable buffering on the writable side.
```

Comment:
I don't believe it is appropriate to remove the write side buffering. It allows transform streams to buffer input in cases where the `_transform()` implementation is slow.

Reply:
If you have a pipeline then the previous stream should buffer in its readable side so I don't think it matters so much. But it is an interesting question, what side do we buffer on?

Reply:
I made it possible to override with `options.writableHighWaterMark` so it's up to the user where to do the buffering. Default behavior is to only buffer on the readable side.